Payment methods: fix alignment of 'Use as backup' text

ramonjd opened the pull request:
Related to #93710



## Proposed Changes

* Removes references to component attributes in CSS declarations.
* Align the `'Delete this payment method'` text in viewports <660px to make it consistent with all widths 
* ~~Ensure that items in the right hand column of the payment methods grid align to the right.~~
* ~~Swap out delete text for trash icon on narrow viewport widths (optional)~~

Checked on Firefox, Chrome and Safari

## Why are these changes being made?

* CSS rules shouldn't target component internals
* ~~The 'Use as backup' text element does not align with the delete text beneath it~~
* ~~The 'Delete this payment method' text bunches up and crowds the space on narrow viewport widths.~~

## Testing Instructions

1. Fire up this branch and head to http://calypso.localhost:3000/me/purchases/payment-methods
2. If you don't have any payment methods added, add some!
3. Check that the 'Use as backup.' text aligns to the right.
4. Narrow the viewport width. Check that the 'Delete this payment method' text aligns right.
5. There should be no other differences between this PR and trunk.


### Before
| <440 | <660 | 1200 |
|--------|--------|--------|
| <img width="384" alt="Screenshot 2024-09-09 at 1 54 38 PM" src="https://github.com/user-attachments/assets/9119617c-65ba-4d23-88e2-75bb5553c334"> | <img width="645" alt="Screenshot 2024-09-09 at 1 56 46 PM" src="https://github.com/user-attachments/assets/1372cea5-a9bc-4fee-8163-a0fad8c1a322"> | <img width="905" alt="Screenshot 2024-09-09 at 1 58 34 PM" src="https://github.com/user-attachments/assets/69529b1d-c7f6-4b75-9395-c5fefced4f77"> | 



### After
| <440 | <660 | 1200 |
|--------|--------|--------|
| <img width="385" alt="Screenshot 2024-09-09 at 1 52 51 PM" src="https://github.com/user-attachments/assets/09e21519-d03f-453c-8efe-2a1effa64330"> | <img width="644" alt="Screenshot 2024-09-09 at 1 55 54 PM" src="https://github.com/user-attachments/assets/9197ab12-7b1c-458e-a27e-c889fccd1c38"> | <img width="903" alt="Screenshot 2024-09-09 at 1 58 30 PM" src="https://github.com/user-attachments/assets/f0879d15-c72f-4e74-a6e9-28660f228d62"> | 











## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

ramonjd committed:
Ensure that items in the right hand column of the grid align to the right.
Swap out delete text for trash icon on narrow viewport widths.

Revert changes from https://github.com/Automattic/wp-calypso/pull/93798

ramonjd commented on client/me/purchases/payment-methods/payment-method-delete.tsx:
```diff
@@ -59,7 +61,7 @@ const PaymentMethodDelete: FunctionComponent< Props > = ( { card } ) => {
 				borderless
 			>
-				{ text }
+				{ isNarrowView ? <Gridicon icon="trash" /> : text }
```
This is optional. I thought it might look nicer in narrow viewports, but it doesn't directly address any issues reported in https://github.com/Automattic/wp-calypso/issues/93710

michaeldcain replied:
In Checkout, we've intentionally moved away from using icons because of their ambiguity, and I think we intentionally expanded the text here in #62576, so that it was explicit that the entire payment method was being deleted (instead of just the billing information). Since the action is on its own line, I don't know that we're restricted by space.

ramonjd replied:
Makes sense. Thanks for the quick feedback. Happy to drop it as it was a cheeky proposal and, as mentioned, doesn't speak to the issue.

ramonjd commented:
Seems to have been addressed in https://github.com/Automattic/wp-calypso/pull/93798 without considering this PR.

michaeldcain reviewed (changes requested):
Thanks for re-opening this issue! I'm sorry it was missed originally :/ I'm going to ping @Automattic/shilling and drop this in our review column to make sure this gets reviewed this week while I'm AFK.

In general this is the correct fix, but I think I'm not sold on the icon alone, and it looks like we've lost some left padding at mobile widths. Maybe we could drop it to a single column at 660px instead of 440px and either leave the action text alone, or add the icon alongside the action text.

| Production      | This PR |
| ----------- | ----------- |
| <img width="347" alt="Screenshot 2024-09-08 at 10 17 50 PM" src="https://github.com/user-attachments/assets/f84af06d-8940-4b2d-a1cf-2e3c5c4abf3f"> | <img width="345" alt="Screenshot 2024-09-08 at 10 17 34 PM" src="https://github.com/user-attachments/assets/f2f01d0a-d7ee-4efc-b87c-0aedd4f4067f"> |

ramonjd committed:
Revert icon
Restore left padding for backup label

ramonjd commented:
> I think I'm not sold on the icon alone, and it looks like we've lost some left padding at mobile widths.

@michaeldcain Thanks for the review. I've fixed the padding and removed the icon.

The screenshots in the description are updated to reflect the difference between this PR and trunk - the only visible difference is the alignment of the delete text (right) in <660px.

michaeldcain reviewed (approved):
Works as described. You're an icon.

ramonjd commented:
> You're an icon.

But not a trash icon. 🤣 

Thanks for testing again!
